.circleci/config.yml

```diff
@@ -223,12 +223,41 @@ jobs:
       - run: ./gradlew --no-daemon --stacktrace integrationTest
       - run:
           when: on_fail
-          command: cat integration/spark/build/test-results/integrationTests/TEST-*.xml
+          command: cat integration/spark/build/test-results/integrationTest/TEST-*.xml
       - run: ./gradlew --no-daemon jacocoTestReport
       - store_test_results:
-          path: integration/spark/build/test-results/integrationTests
+          path: integration/spark/build/test-results/integrationTest
       - store_artifacts:
-          path: integration/spark/build/reports/tests/integrationTests
+          path: integration/spark/build/reports/tests/integrationTest
+          destination: test-report
+      - save_cache:
+          key: v1-integration-spark-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - ~/.gradle
+
+  test-integration-spark-v3:
```

Comment:
Can we define a [matrix job](https://circleci.com/blog/circleci-matrix-jobs) to handle different spark versions?

Reply:
nice, I guess we can do it.